### ISSUE TITLE
[Bug] Ubuntu CKAN crashes on launch

### ISSUE THREAD

**Issue by hinneLinks** [KSP-CKAN/CKAN#3046]
Background
----------
* Operating System: Ubuntu 18.04
* CKAN Version: 1.27.0, 1.27.2
* KSP Version: 1.9.1.2788
* Mono: 6.8.0.105

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?**
No

Problem
-------
The first start of a day, CKAN crashes. Every subsequent start works normally.
It seems, the error occurs, while CKAN is trying to ask me something: `CKAN.GUIUser:RaiseYesNoDialog`
Might be related to https://github.com/KSP-CKAN/CKAN/issues/3015 

I've tried 1.27.2 (installed it via downloading it manually), used on the same day when i used 1.27.0 it works, but on the next day it spinning forever searching for updates until i force shut it down after a couple of minutes (no logs in the terminal). I've downgraded to 1.27.0 again to be able to use CKAN.

**Steps to reproduce**
Start CKAN the first time of a Day

**Expected behavior**
No Crashes

**Screenshots (if applicable)**
n/a
**CKAN error code (if applicable):**

```
	  at <unknown> <0xffffffff>
	  at System.Windows.Forms.X11Keyboard:XCreateFontSet <0x000d6>
	  at System.Windows.Forms.X11Keyboard:CreateOverTheSpotXic <0x000e7>
	  at System.Windows.Forms.X11Keyboard:CreateXic <0x00117>
	  at System.Windows.Forms.X11Keyboard:CreateXicForWindow <0x00053>
	  at System.Windows.Forms.X11Keyboard:FocusIn <0x000c7>
	  at System.Windows.Forms.XplatUIX11:SetFocus <0x000f3>
	  at System.Windows.Forms.XplatUI:SetFocus <0x0003c>
	  at System.Windows.Forms.ContainerControl:SendControlFocus <0x000c7>
	  at System.Windows.Forms.Form:SetVisibleCore <0x00517>
	  at System.Windows.Forms.Control:set_Visible <0x0004b>
	  at System.Windows.Forms.Control:set_Visible <0x00083>
	  at System.Windows.Forms.Application:RunLoop <0x003df>
	  at System.Windows.Forms.Form:ShowDialog <0x0091f>
	  at System.Windows.Forms.Form:ShowDialog <0x0002b>
	  at CKAN.YesNoDialog:ShowYesNoDialog <0x001cf>
	  at CKAN.YesNoDialog:ShowYesNoDialog <0x000a3>
	  at CKAN.Main:YesNoDialog <0x00057>
	  at CKAN.Main:YesNoDialog <0x000af>
	  at CKAN.GUIUser:RaiseYesNoDialog <0x0003f>
	  at CKAN.Repo:HandleModuleChanges <0x0019f>
	  at CKAN.Repo:UpdateAllRepositories <0x0051b>
	  at CKAN.Main:UpdateRepo <0x005cb>
	  at System.ComponentModel.BackgroundWorker:OnDoWork <0x00046>
	  at System.ComponentModel.BackgroundWorker:WorkerThreadStart <0x000c2>
	  at System.ComponentModel.BackgroundWorker:<RunWorkerAsync>b__27_0 <0x00033>
	  at System.Threading.Tasks.Task:InnerInvoke <0x00060>
	  at System.Threading.Tasks.Task:Execute <0x00031>
	  at System.Threading.Tasks.Task:ExecutionContextCallback <0x00045>
	  at System.Threading.ExecutionContext:RunInternal <0x00147>
	  at System.Threading.ExecutionContext:Run <0x00022>
	  at System.Threading.Tasks.Task:ExecuteWithThreadLocal <0x000e9>
	  at System.Threading.Tasks.Task:ExecuteEntry <0x00156>
	  at System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem <0x00014>
	  at System.Threading.ThreadPoolWorkQueue:Dispatch <0x0028a>
	  at System.Threading._ThreadPoolWaitCallback:PerformWaitCallback <0x00012>
	  at <Module>:runtime_invoke_bool <0x00086>
```

More Details see attached [ckan-error.log](https://github.com/KSP-CKAN/CKAN/files/4588490/ckan-error.log).

**Comment by HebaruSan** [comment]
> It seems, the error occurs, while CKAN is trying to ask me something: `CKAN.GUIUser:RaiseYesNoDialog`

Can you please share the full text from which you got that string? That should give us clues as to what's going on.

**Comment by HebaruSan** [comment]
The stack trace is identical to #3027, which is surprising after #3006 changed how this code works.

@hinneLinks , can you double check that you are running the versions that you think you are? If you have two copies of ckan.exe and one works and one does not, I would expect the one that does not work to be 1.27.0 and the one that does work to be 1.27.2 since we previously had this exact issue reported on 1.27.0 and made changes to fix it in 1.27.2.

**Comment by hinneLinks** [comment]
The Stacktrace is from Version 1.27.0, not 1.27.2.
I've installed 1.27.2 again and waited longer, now i got a dialog asking me to reinstall some mods due changed metadata. 
So Updating (and waiting) did the trick, thank you.
